### PR TITLE
Add a test of sending primitive values for a resource

### PR DESCRIPTION
**Pull request by Frassle**
We already had the "simple" provider test that just checked we could make a resource with a single simple bool field. This adds a new "primitive' provider and checks we can send strings, numbers, integers, arrays and maps from the program.

### CODE REVIEW

File: sdk/go/pulumi-language-go/language_test.go [hunk 1]
```diff
@@ -187,6 +187,7 @@ var expectedFailures = map[string]string{
 	"l2-resource-asset-archive":          "missing go.mod",
 	"l2-failed-create-continue-on-error": "missing go.mod",
 	"l2-engine-update-options":           "missing go.mod",
+	"l2-resource-primitives":             "missing go.mod",
```
[the remark this answers is not in the document]
Comment (Frassle) [reply sdk/go/pulumi-language-go/language_test.go:190]: Programgen and sdkgen are broken right now

File: cmd/pulumi-test-language/providers/primitive_provider.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+				Type: t,
+			},
+		}
+	}
+	resourceProperties := map[string]schema.PropertySpec{
+		"b": makePrimitive("boolean"),
+		"f": makePrimitive("number"),
+		"i": makePrimitive("integer"),
+		"s": makePrimitive("string"),
+		"a": {
```
[the remark this answers is not in the document]
Comment (Frassle) [reply cmd/pulumi-test-language/providers/primitive_provider.go:66]: Sure

**Comment by Frassle** [comment]
> Is it worth removing the simple provider and using this in its stead? 

No I want to keep that around as a really really simple sdk to generate and runtime against so that you've got something as simple as possible to target when bringing up a new language.

**Comment by lunaris** [comment]
> > Is it worth removing the simple provider and using this in its stead?
> 
> No I want to keep that around as a really really simple sdk to generate and runtime against so that you've got something as simple as possible to target when bringing up a new language.

I wonder if it's worth adding a doc to `simple` (not necessarily here) to make that clear too. I'm guessing that at least one property is required to make it the simplest possible? Or are zero-propertied permissible? Just spitballing, happy either way!

**Comment by Frassle** [comment]
One property is needed to check we can actually generate and send args. I'll raise a PR to add that comment.

**Pull request merged**